packages/data/src/options/resolvers.js

```diff
@@ -19,7 +19,32 @@ export function* getOptionsWithRequest( names ) {
 		yield receiveOptions( results );
 		return results;
 	} catch ( error ) {
-		yield setRequestingError( error );
+		yield setRequestingError( error, names );
 		return error;
 	}
 }
+
+const fetches = {};
+
+export function* getOption( name ) {
+	yield setIsRequesting( name );
+	const names = yield getOptionsToRequest( name );
+
+	const fetchInProgress = fetches[ names ];
+	if ( fetchInProgress ) {
```

Comment:
You could probably just eliminate the declaration of `fetchInProgress` here and use `fetches[names]` directly in the condition.